src/io.js

```diff
@@ -284,76 +284,118 @@ export function JSONP(opt) {
 JSONP.counter = 0;
 JSONP.requests = {};
 
-// Load multiple Image assets async
-export function Images(opt) {
-  opt = $.merge({
-    src: [],
-    noCache: false,
-    onProgress: $.empty,
-    onComplete: $.empty
-  }, opt || {});
-
-  let count = 0;
-  let l = opt.src.length;
-
-  let images;
-  // Image onload handler
-  var load = () => {
-    opt.onProgress(Math.round(++count / l * 100));
-    if (count === l) {
-      opt.onComplete(images);
-    }
-  };
-  // Image error handler
-  var error = () => {
-    if (++count === l) {
-      opt.onComplete(images);
-    }
-  };
-
-  // uid for image sources
-  const noCache = opt.noCache;
-  const uid = $.uid();
-  function getSuffix(s) {
-    return (s.indexOf('?') >= 0 ? '&' : '?') + uid;
-  }
-
-  // Create image array
-  images = opt.src.map((src, i) => {
-    const img = new Image();
-    img.index = i;
-    img.onload = load;
-    img.onerror = error;
-    img.src = src + (noCache ? getSuffix(src) : '');
-    return img;
+// Creates an image-loading promise.
+function loadImage(src) {
+  return new Promise(function(resolve, reject) {
+    var image = new Image();
+    image.onload = function() {
+      resolve(image);
+    };
+    image.onerror = function() {
+      reject(new Error(`Could not load image ${src}.`));
+    };
+    image.src = src;
   });
+}
 
-  return images;
+// Load multiple images async.
+// rye: TODO this needs to implement functionality from the
+//           original Images function.
+async function loadImages(srcs) {
+  let images = srcs.map((src) => loadImage(src));
+  let results = [];
+  for (let image of images) {
```

Comment:
Perhaps use naming to be a little clearer about what is going on?

   for (const imagePromise of imagePromises) ?